client/my-sites/checkout/composite-checkout/hooks/use-remove-from-cart-and-redirect.ts

```diff
@@ -23,6 +25,9 @@ export default function useRemoveFromCartAndRedirect(
 	removeProductFromCartAndMaybeRedirect: RemoveProductFromCart;
 } {
 	const { removeProductFromCart } = useShoppingCart();
+	// The cloud.jetpack.com/pricing page sends a `checkoutBackUrl` url query param to checkout.
+	const { checkoutBackUrl } = useSelector( getInitialQueryArguments ) as { [ k: string ]: string };
```

Comment:
`getInitialQueryArguments` has poor type documentation, but it can also return `null`, which might cause this line to break (and the `as` here hides the issue further). My suggestion is to change this line to:

```suggestion
	const { checkoutBackUrl } = useSelector( getInitialQueryArguments ) ?? {};
```

And then apply this diff to `client/state/selectors/get-initial-query-arguments.js`:

```diff
diff --git a/client/state/selectors/get-initial-query-arguments.js b/client/state/selectors/get-initial-query-arguments.js
index 6e1e6aa7e3..d922dab482 100644
--- a/client/state/selectors/get-initial-query-arguments.js
+++ b/client/state/selectors/get-initial-query-arguments.js
@@ -12,7 +12,7 @@ import 'calypso/state/route/init';
  * Gets the initial query parameters set by a ROUTE_SET action
  *
  * @param {object} state - global redux state
- * @returns {object} initial query arguments
+ * @returns {Object.<string, string>|null} initial query arguments
  */
 export const getInitialQueryArguments = ( state ) => get( state, 'route.query.initial', null );
```